crates/snp_validation/src/config.rs

```diff
@@ -0,0 +1,148 @@
+
+    /// Minimum value for `policy.abi_major`.`policy.abi_minor`
+    #[serde(default)]
+    pub abi: VersionReq,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_policy_string")]
+    pub policy_flags: Option<u8>,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_platform_string")]
+    pub platform_info_flags: Option<PlatformInfoFlags>,
+}
+
+fn from_policy_string<'de, D>(deserializer: D) -> Result<Option<u8>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: Vec<SnpPolicyFlags> = Deserialize::deserialize(deserializer)?;
+
+    let mut flags = FlagSet::<PolicyFlags>::from(PolicyFlags::Reserved);
+
+    for flag in s {
+        match flag {
+            SnpPolicyFlags::Debug => {
+                flags |= PolicyFlags::Debug;
+            }
+            SnpPolicyFlags::SingleSocket => {
+                flags |= PolicyFlags::SingleSocket;
+            }
+            SnpPolicyFlags::SMT => {
+                flags |= PolicyFlags::SMT;
+            }
+        }
+    }
+
+    Ok(Some(flags.bits()))
+}
+
+fn from_platform_string<'de, D>(deserializer: D) -> Result<Option<PlatformInfoFlags>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let flags: SnpPlatformInfoFlags = Deserialize::deserialize(deserializer)?;
+
+    Ok(Some(match flags {
+        SnpPlatformInfoFlags::SME => PlatformInfoFlags::SME,
+        SnpPlatformInfoFlags::TSME => PlatformInfoFlags::TSME,
+    }))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::config::Config;
+
+    #[test]
+    fn empty_config() {
+        let result: Result<Config, toml::de::Error> = toml::from_str("");
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn flags() {
+        let config: Config = toml::from_str(
+            r#"
+        enarx_signer = ["1234567890", "00112233445566778899"]
+        policy_flags = ["SingleSocket", "Debug"]
+        platform_info_flags = "SME"
+        "#,
+        )
+        .expect("Couldn't deserialize");
+        assert!(config.policy_flags.is_some());
+        assert!(config.platform_info_flags.is_some());
+
+        let flags = FlagSet::<PolicyFlags>::new(config.policy_flags.unwrap()).unwrap();
+        assert_eq!(
+            flags,
+            PolicyFlags::SingleSocket | PolicyFlags::Debug | PolicyFlags::Reserved
+        );
+
+        assert_eq!(config.measurements.signer.len(), 2);
+        assert_eq!(
+            config.measurements.signer.get(0).unwrap().as_ref(),
+            &hex::decode("1234567890").unwrap()
+        );
+        assert_eq!(
+            config.measurements.signer.get(1).unwrap().as_ref(),
+            &hex::decode("00112233445566778899").unwrap()
+        );
+    }
+
+    #[test]
+    fn semver() {
+        let config: Config = toml::from_str(
+            r#"
+        enarx_signer = ["1234567890", "00112233445566778899"]
+        abi = "1.0"
+        "#,
+        )
+        .expect("Couldn't deserialize");
+        assert_eq!(config.abi.to_string(), "^1.0");
```

Comment:
you should assert the `enarx_signer` field here as well

Reply:
Fixed